### PR TITLE
echo: reworking of the new options

### PR DESCRIPTION
**Pull request by alstolten**
Unfortunately https://docs.rs/clap/2.33.3/clap/struct.Arg.html#method.allow_hyphen_values does not fix this. I played around a little bit, but with no success. I reworked #1884 a little bit.

I added a failing test case for leading hyphens in echo.

With C implementation echo -foo works fine.

### CODE REVIEW

**Comment by sylvestre** [comment]
Seems that we have a fix here:
https://github.com/uutils/coreutils/pull/1887
@alstolten is that ok with you?

**Comment by ycd** [comment]
@sylvestre, i think we can merge both PRs in, this looks like a refactor of #1884.

**Comment by sylvestre** [comment]
@alstolten 
could you please rebase your patch?

**Comment by alstolten** [comment]
@sylvestre 
Will do.

**Comment by alstolten** [comment]
@sylvestre 
Not sure if this has the correct format now. If not please give me a hint.

File: src/uu/echo/src/echo.rs [hunk 1]
```diff
@@ -122,34 +122,41 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
         // Final argument must have multiple(true) or the usage string equivalent.
         .setting(clap::AppSettings::TrailingVarArg)
         .setting(clap::AppSettings::AllowLeadingHyphen)
-        .version(VERSION)
-        .usage(SYNTAX)
+        .version(crate_version!())
```
Comment (sylvestre) [inline src/uu/echo/src/echo.rs:125]: we could do that in other different places in coreutils. Would you be interested in doing it? thanks

File: src/uu/echo/src/echo.rs [hunk 2]
```diff
@@ -122,34 +122,41 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
...
+        .usage(USAGE)
```
Comment (sylvestre) [inline src/uu/echo/src/echo.rs:126]: needs rustfmt here ;)

File: src/uu/echo/src/echo.rs (hunk 1 above)
Comment (alstolten) [reply src/uu/echo/src/echo.rs:125]: I'll check

**Comment by sylvestre** [comment]
the test was interesting too, no?

**Comment by alstolten** [comment]
> 
> 
> the test was interesting too, no?

@ycd has already done a test in https://github.com/uutils/coreutils/pull/1887

File: src/uu/echo/src/echo.rs (hunk 1 above)
Comment (sylvestre) [reply src/uu/echo/src/echo.rs:125]: thanks!

File: src/uu/echo/src/echo.rs [hunk 3]
```diff
@@ -9,17 +9,17 @@
 #[macro_use]
 extern crate uucore;
 
-use clap::{App, Arg};
+use clap::{App, Arg, crate_version};
```
Comment (sylvestre) [inline src/uu/echo/src/echo.rs:12]: please run rustfmt

File: src/uu/echo/src/echo.rs (hunk 2 above)
Comment (alstolten) [reply src/uu/echo/src/echo.rs:126]: Ok, it should be ok now for rustfmt

**Comment by alstolten** [comment]
@sylvestre 
Thanks for the help, I did a rustfmt run.

**Pull request merged**